client/src/components/ChatComponent/index.js

```diff
@@ -0,0 +1,102 @@
+/* eslint-disable object-curly-newline */
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Layout, Icon, Avatar, Button, Input, Skeleton, List } from 'antd';
+
+const { Content } = Layout;
+const IconText = ({ type, text }) => (
+  <span>
+    <Icon type={type} style={{ marginRight: 8 }} />
+    {text}
+  </span>
+);
+IconText.propTypes = {
+  type: PropTypes.string.isRequired,
+  text: PropTypes.string.isRequired,
+};
+
+const listData = [];
+for (let i = 0; i < 3; i += 1) {
+  listData.push({
+    href: '#',
+    title: `username ${i}`,
+    avatar: 'https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png',
+    content:
+      'We supply a series of design principles, practical patterns and high quality design resources (Sketch and Axure), to help people create their product prototypes beautifully and efficiently.',
+  });
+}
+
+const ChatComponent = () => {
+  const [loading, setLoading] = useState(true);
+  const [messages] = useState(listData);
+  useEffect(() => {
+    setTimeout(() => setLoading(false), 2000);
+  }, []);
+  function sendMessage(e) {
+    console.log('sendingMessage', e.target.value);
+    e.target.value = '';
+  }
+  return (
+    <Content
+      style={{
+        background: '#fff',
+        padding: 24,
+        margin: 0,
+        minHeight: 280,
+      }}
+    >
+      <div className="chat-box">
+        <div className="chat-history">
+          <div>
+            <List
+              itemLayout="vertical"
+              size="large"
+              dataSource={messages}
+              renderItem={item => (
+                <List.Item
+                  key={item.title}
+                  actions={
+                    !loading && [
+                      <IconText type="star-o" text="156" />,
+                      <IconText type="like-o" text="156" />,
+                      <IconText type="message" text="2" />,
+                    ]
+                  }
+                >
+                  <Skeleton loading={loading} active avatar>
+                    <List.Item.Meta
+                      avatar={<Avatar src={item.avatar} />}
+                      title={<a href={item.href}>{item.title}</a>}
+                    />
+                    {item.content}
+                  </Skeleton>
+                </List.Item>
+              )}
+            />
+          </div>
+        </div>
+        <div className="chat-input">
+          <Input
+            onPressEnter={sendMessage}
+            placeholder="Say it"
+            size="large"
+            prefix={<Icon type="user" style={{ color: 'rgba(0,0,0,.25)' }} />}
+            suffix={(
+              <>
+                <Button shape="circle" size="medium">
```

Comment:
Create a generic button component, so that it is consistent among other components.

Reply:
Updated